cellprofiler/module.py

```diff
@@ -876,3 +877,77 @@ def on_deactivated(self):
     def on_setting_changed(self, setting, pipeline):
         '''Called when a setting has been changed in the GUI'''
         pass
+
+
+class ImageProcessing(Module):
+    category = "Image Processing"
+
+    def create_settings(self):
+        self.x_name = cellprofiler.setting.ImageNameSubscriber(
+            "Input"
+        )
+
+        self.y_name = cellprofiler.setting.ImageNameProvider(
+            "Output",
+            self.__class__.__name__
+        )
+
+    def display(self, workspace, figure):
+        layout = (2, 1)
+
+        figure.set_subplots(
+            dimensions=workspace.display_data.dimensions,
+            subplots=layout
+        )
+
+        figure.subplot_imshow(
+            dimensions=workspace.display_data.dimensions,
+            image=workspace.display_data.x_data,
+            x=0,
+            y=0
+        )
+
+        figure.subplot_imshow(
+            dimensions=workspace.display_data.dimensions,
+            image=workspace.display_data.y_data,
+            x=1,
+            y=0
+        )
+
+    def run(self, workspace):
+        x_name = self.x_name.value
+
+        y_name = self.y_name.value
+
+        images = workspace.image_set
+
+        x = images.get_image(x_name)
+
+        dimensions = x.dimensions
+
+        x_data = x.pixel_data
+
+        args = (setting.value for setting in self.settings()[2:])
+
+        y_data = self.function(x_data, *args)
```

Comment:
Interesting! I like where this is going.

I think that `args` should be set from `self.visible_settings`, for the case where there are multiple operations per module. 

Additionally, how do we add arguments like `multichannel`, which are required but not determined from a setting? I am not sure how many of the strictly "Image Processing" modules use a `multichannel` option, so let's wait until we need to implement this.